### ISSUE TITLE
MD025 is passing if plain text is above the first h1

### ISSUE THREAD

**Issue by pityonline** [markdownlint/markdownlint#292]
## Description

MD025 is passing if plain text is above the first h1.

### Environment

**MDL Version** 0.5.0

### Expected Behavior

Multiple h1 titles should be detected.

### Actual Behavior

MD025 passed.

## Replication Case

1. Create a text file `foo.md`:

```
text above the first h1

# title 1

text

# title 2

text
```

2. Run `mdl -v -r MD025 foo.md`

```
Checking bar.md...
Processing rule MD025
```

**Comment by jaymzh** [comment]
This is expected. As the docs say:

"This rule is triggered when a top level header is in use (the first line of the file is a h1 header), and more than one h1 header is in use in the document:"

Only when an H1 is being used as a top-level header (i.e. the first line) are **and** there are also multiple H1s. As the rational on the RULES.md states, the idea is that **if** yuu are using H1 as a Document Header as opposed to just the biggest section header, there should be only one, since there's only one document.
